rtengine/boxblur.h

```diff
@@ -312,6 +313,222 @@ template<class T, class A> void boxblur (T** src, A** dst, T* buffer, int radx,
 
 }
 
+inline void boxblur (float** src, float** dst, int radius, int W, int H, bool multiThread)
+{
+    //box blur using rowbuffers and linebuffers instead of a full size buffer
+
+    if (radius == 0) {
+        if (src != dst) {
+#ifdef _OPENMP
+            #pragma omp parallel for if (multiThread)
+#endif
+
+            for (int row = 0; row < H; row++) {
+                for (int col = 0; col < W; col++) {
+                    dst[row][col] = src[row][col];
+                }
+            }
+        }
+        return;
+    }
+
+    constexpr int numCols = 8; // process numCols columns at once for better usage of L1 cpu cache
+#ifdef _OPENMP
+    #pragma omp parallel if (multiThread)
+#endif
+    {
+        float* const buffer = new float[std::max(W, 8 * H)];
```

Comment:
Should we still allocate raw buffers on heap in new/refactored code? Would using a `std::vector<float>` hurt?

Reply:
I used raw buffers here because I later reinterprete them e.g. as vfloat buffers.
I dont know how's the best way to do this using std::vector<float>, std::vector<vfloat>.
Maybe you can help on this?

Reply:
Well, at least, it could be a `std::unique_ptr<>`...

Regarding `std::vector<>`: it has gained a [`data()`](https://en.cppreference.com/w/cpp/container/vector/data) function, which should be the one you're after.

Reply:
Changed to std::unique_ptr<>